src/core/public/application/utils.ts

```diff
@@ -52,3 +55,55 @@ export const appendAppPath = (appBasePath: string, path: string = '') => {
     leading: false,
   });
 };
+
+export function isLegacyApp(app: App | LegacyApp): app is LegacyApp {
+  return app.legacy === true;
+}
+
+/**
+ * Converts a relative path to an absolute url.
+ * Implementation is based on a specified behavior of the browser to automatically convert
+ * a relative url to an absolute one when setting the `href` attribute of a `<a>` html element.
+ *
+ * @example
+ * ```ts
+ * // current url: `https://kibana:8000/base-path/app/my-app`
+ * relativeToAbsolute('/base-path/app/another-app') => `https://kibana:8000/base-path/app/another-app`
+ * ```
+ */
+export const relativeToAbsolute = (url: string): string => {
+  const a = document.createElement('a');
+  a.setAttribute('href', url);
+  return a.href;
+};
+
+/**
+ * Return the current origin of the browser
+ */
+export const getOrigin = (): string => {
+  return window.location.origin;
+};
+
+/**
+ * Return true if given element or any parent has the given class.
+ */
+export const selfOrParentHasClass = (element: HTMLElement, className: string): boolean => {
+  let current = element;
+  while (current) {
+    if (current.classList.contains(className)) {
+      return true;
+    }
+    if (!current.parentElement || current.parentElement === document.body) {
+      break;
+    }
+    current = current.parentElement;
+  }
+  return false;
+};
+
+/**
+ * Return true if any modifier key is pressed.
+ */
+export const isModifiedEvent = (event: MouseEvent): boolean => {
+  return !!(event.metaKey || event.altKey || event.ctrlKey || event.shiftKey);
+};
```

Comment:
Did not add tests for that as it is rather trivial and forging synthetic events is a hassle.